### DISCUSSION TITLE
How to use `sea-orm` in a Rust library project?

### DISCUSSION THREAD

**Discussion by mainrs** [SeaQL/sea-orm#1684] [Q&A]
Hello!

I need a `sqlite` database for one of my own libraries and wanted to use `sea-orm` instead of `sqlx` directly. However, since `sea-orm` uses different crates (`migration`, `entity`, `application`), publishing the library to crates-io is not possible anymore.

Is there a way to combine migrations, entities and the actual library code into a single crate that I can then publish? I would rather not publish three different crates, two of which cannot really be used on their own (`migration` and `entities`).

Thanks!

**Comment by billy1624** [comment]
Hey @mainrs, that's possible. So, you have a single crate with `migration`, `entity`, `application` modules.
